ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committing/ProcessTransactionSubmission.scala

```diff
@@ -208,6 +212,63 @@ private[kvutils] case class ProcessTransactionSubmission(
           buildRejectionLogEntry(RejectionReason.Disputed("DuplicateKey: Contract Key not unique")))
     } yield r
 
+  /** Check that all parties mentioned in a transaction exist. */
+  private def validateParties: Commit[Unit] = {
+
+    def foldPartiesInTx[T](tx: GenTransaction.WithTxValue[NodeId, TContractId], z: T)(f: (T, String) => T): T =
+      foldPartiesInTxActors(tx, foldPartiesInTxVals(tx, z)(f))(f)
+
+    def foldPartiesInTxActors[T](tx: GenTransaction.WithTxValue[_, _], z: T)(f: (T, String) => T): T =
+      tx.fold(z) {
+        case (accum, (_, n)) => {
+          val parties = n match {
+            case c: NodeCreate[_, _] =>
+              (c.signatories union c.stakeholders)
```

Comment:
Don't the stakeholders already include the signatories?

Reply:
Yes it does, nice catch (leftover of wrong copy-over of `signatories union observers` elsewhere).